src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -137,17 +138,17 @@
     /**
      * Our source of reference data (null if no reference was provided)
      */
-    ReferenceDataSource reference;
+    protected ReferenceDataSource reference;
```

Comment:
Finally these fields get protected. I was looking forward to access the data sources and I was planning to do a PR for this. But it won't be better to have a getter that throws if `null`? This will prevent users of `GATKTool` without `require*` methods returning `true` to use a field that isn't set (and get rid of implementation discrepancies...)

Reply:
I like the idea, but I think those methods would have to query the `require*` methods to see if they should throw anyway.  I'd prefer to get this in and then fix the interfaces at a later. 

@magicDGS - can you add fixing the interfaces to these objects as an issue?